### PR TITLE
Detect and fix gtest and gmock dependencies.

### LINKED ISSUE #1219: Catch gtest headers in fix_cc_deps
Need to autofix #1218

### PR DESCRIPTION
**Pull request by jonmeow**
Fixes #1219 

The repo is correct, I'm just trying to avoid regressions from this.

### CODE REVIEW

File: explorer/testdata/operators/add_builtin.carbon [hunk 1]
```diff
@@ -11,9 +11,6 @@
 
 package ExplorerTest api;
 
-// TODO: T:! Add
-fn DoAdd[T:! AddWith(.Self) where .Result == .Self](x: T, y: T) -> T { return x + y; }
-
```
Comment (chandlerc) [inline explorer/testdata/operators/add_builtin.carbon]: I think this is an unrelated change due to needing to merge up something...

**Review by chandlerc — CHANGES_REQUESTED** [review]
Needs a merge fix, but then LG.

File: explorer/testdata/operators/add_builtin.carbon (hunk 1 above)
Comment (jonmeow) [reply explorer/testdata/operators/add_builtin.carbon]: Fixed

**Review by chandlerc — APPROVED** [review]
